### PR TITLE
feat(editor): Download service charge CSV

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Wires up `.csv` download from service charges page
 - Adds Storybook coverage

Storybook demo here - https://storybook.7086.planx.pizza/?path=/docs/editor-components-subscription-servicecharges--docs

Trello ticket - https://trello.com/c/FJjck1KD/3680-fix-download-service-charge-payment-records-csv-link-in-editor

### CODE REVIEW

File: apps/api.planx.uk/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "cookie-session": "^2.1.1",
     "copyfiles": "^2.4.1",
     "cors": "^2.8.6",
-    "csv-stringify": "^6.8.0",
```
Comment (joecarver) [inline apps/api.planx.uk/package.json:28]: nice! not seen this before but seems handy

File: apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/utils.ts [hunk 2]
```diff
@@ -35,3 +36,31 @@ export const formatUKFiscalYear = (fiscalYear: number): string => {
...
+  serviceCharges: ServiceCharge[],
+): string =>
+  stringify(serviceCharges, {
+    header: true,
+    columns: [
+      { key: "flowName", header: "Flow name" },
+      { key: "sessionId", header: "Session ID" },
+      { key: "paymentId", header: "Payment ID" },
+      { key: "amount", header: "Amount (excl VAT)" },
+      { key: "paidAt", header: "Paid at" },
```
Comment (joecarver) [inline apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/utils.ts:50]: nit: only potential issue i can think of is that some of these values could intefer with csv parsing if they contained commas or even more special characters that are interpreted as formulas/rules in excel? i guess it would only be flowName as that's the only customisable value, but seems likely it could contain a comma in some cases at least? so might be worth sanitising it here.

File: apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/components/ServiceCharges.tsx [hunk 3]
```diff
@@ -96,7 +97,15 @@ const ActiveServiceCharges = ({ serviceCharges }: SubscriptionProps) => {
...
-        <Link component="button" onClick={() => console.log("todo")}>
+        <Link
+          component="button"
+          onClick={() =>
+            downloadServiceChargesCsv(
+              serviceChargesInActiveYear,
+              `planx-service-charges-${activeYear}.csv`,
+            )
+          }
+        >
```
Comment (joecarver) [inline apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/components/ServiceCharges.tsx:109]: nit: do we want to style this like an actual button? or are we happy just with the plain text style link?

**Review by joecarver — APPROVED** [review]
Looks good and working well! couple of v minor suggestions but nothing blocking.

Also just occurred to me that maybe we want to add service_charges to the sync script so we have data to test this on pizza and local?

File: apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/utils.ts (hunk 2 above)
Comment (DafyddLlyr) [reply apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/utils.ts:50]: I've been caught out by this before too many times 😅 This is why I went for `csv-stringify` which handles escape characters etc instead of a hand-rolled implementation.

I'll add test coverage here though for this - if we swap or remove the library but don't handle this edge case we want something to break.

Comment (DafyddLlyr) [reply apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/utils.ts:50]: Please see [c06870f](https://github.com/theopensystemslab/planx-new/pull/7086/commits/c06870f69ad93ddadb76d3ac9e4aafe120cb4dfa)

File: apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/components/ServiceCharges.tsx (hunk 3 above)
Comment (DafyddLlyr) [reply apps/editor.planx.uk/src/pages/FlowEditor/components/Subscription/components/ServiceCharges.tsx:109]: Good catch, please see [5055455](https://github.com/theopensystemslab/planx-new/pull/7086/commits/5055455ded21153cd9b703431dae4b9a896ec0a3)

image

**Review by jessicamcinchak — COMMENTED** [review]
Thanks for picking this one up!
